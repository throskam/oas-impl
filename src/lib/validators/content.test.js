const createContentValidator = require('./content')

describe('Empty media type', () => {
  const content = {
    'application/json': {}
  }

  const validator = createContentValidator(content)

  it('should return no errors', () => {
    const payload = { value: 1, mediaType: 'application/json' }
    expect(validator(payload)).toMatchSnapshot()
  })
})

describe('Media type', () => {
  const content = {
    'application/json': {
      schema: {
        type: 'integer'
      }
    }
  }

  const validator = createContentValidator(content)

  it('should return no errors when the the value and media type are valid', () => {
    const payload = { value: 1, mediaType: 'application/json' }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is invalid', () => {
    const payload = { value: 'string', mediaType: 'application/json' }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return an unknown media type error when the media type is undefined', () => {
    expect(validator()).toMatchSnapshot()
  })

  it('should return an unsupported media type error when the media type is not supported', () => {
    const payload = { value: 1, mediaType: 'foo/bar' }
    expect(validator(payload)).toMatchSnapshot()
  })
})
