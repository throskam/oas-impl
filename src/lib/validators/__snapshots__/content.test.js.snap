// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty media type should return no errors 1`] = `Array []`;

exports[`Media type should return a validation error when the value is invalid 1`] = `
Array [
  Object {
    "ajv": Object {
      "dataPath": "",
      "keyword": "type",
      "message": "should be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "content[application/json].schema",
    "rule": "schema-invalid",
    "value": "string",
  },
]
`;

exports[`Media type should return an unknown media type error when the media type is undefined 1`] = `
Array [
  Object {
    "path": "content",
    "rule": "content-media-type-required",
    "value": undefined,
  },
]
`;

exports[`Media type should return an unsupported media type error when the media type is not supported 1`] = `
Array [
  Object {
    "path": "content",
    "rule": "content-media-type-unsupported",
    "value": "foo/bar",
  },
]
`;

exports[`Media type should return no errors when the the value and media type are valid 1`] = `Array []`;
